FEAT: 공통 레이아웃 구현

spare8433 opened the pull request:
공통 레이아웃에 해당하는 header, nav(모바일 버전, 일반 버전) 구현 및 관련된
공통 ui 컴포넌트와 사용자 정보 store 등을 추가

spare8433 committed:
FEAT: 공통 레이아웃 구현
공통 레이아웃에 해당하는 header, nav(모바일 버전, 일반 버전) 구현 및 관련된
공통 ui 컴포넌트와 사용자 정보 store 등을 추가

ban-dal committed:
STYLE: eslint, prettier

- prettier 설정 추가

ban-dal committed:
CHORE: .vscode settings

- 추천 extension 추가
- cspell 허용 단어 추가

ban-dal committed:
REFACTOR: code review feedback

* 기본 색상 -> 색상 토큰으로 변경 필요
- ui 컴포넌트 사용 개선
- 불필요한 css 요소 제거

spare8433 committed:
STYLE: Adapt prettier format
tailwind.config.ts 파일 prettier 적용

spare8433 committed:
STYLE: Add vscode setting

settings.json 업데이트

spare8433 committed:
REFACTOR: Change layout code structure

- 레이아웃 관련 일부 컴포넌트 분할
- button, avatar ui 컴포넌트 스타일 중 rounded-full(border-radius:9999px) 인 경우 border-radius 속성의 transition 이 제대로 적용되지 않는 버그로 인해 rounded-full -> rounded-[100%] 로 수정
  -(https://stackoverflow.com/questions/73750210/why-doesnt-my-border-radius-transition-using-tailwind-apply)

spare8433 commented:
@kdh379 리팩토링한 부분 추가했습니다.

ban-dal commented:
group-accordion-menu.tsx 24 Line
여기 "관리자 메뉴" 색상을 text-muted로 해서 색상 대비가 안좋아서 text-muted-foreground로 변경 필요

group-icon.tsx 26 line
align "start"로 되어있어서 tooltip이 위쪽에 나오던데 center가 나을 듯 함

sign.tsx 10 line
여기 tailwindcss를 동적으로 사용하려고 미리 classname을 선언해둔건 좋았는데
필요한 class만 해두는게 나을듯
const size = {
sm: "size-12"
md: "size-16"
}
이런식으로

spare8433 committed:
REFACTOR: Change some layout components

코드 리뷰내용 적용
- group-accordion-menu.tsx: "관리자 메뉴" 색상 변경 및 일부 스타일 변경
- group-icon.tsx: tooltip 위치 정렬을 "start" -> "center" 로 변경
- sign.tsx: 기존 "size" 옵션을 필요한 사이즈들로만 축소(sm: "size-12", md: "size-16")

spare8433 commented:
@kdh379 변경내용 적용했습니다.

ban-dal reviewed (approved):
고생했어.
